src/Jobs/RegisterWebhooks.php

```diff
@@ -16,22 +16,26 @@ public function handle() {
         try {
             /** @var Connection $client */
             $connection = ConnectionFactory::getConnection();
-            // If the connection is not authenticated, don't attempt to register webhooks. Otherwise, this would
-            // break the entire application (including the authenticate-exact endpoint), since this method is
-            // called for every request the user makes.
-            if($connection->needsAuthentication()) {
-                return;
-            }
 
             $webhook = new WebhookSubscription($connection);
 
-            $stockSubscription = collect($webhook->get())->firstWhere('Topic', 'StockPositions');
+            $stockSubscription = collect($webhook->get())
+                ->where('Topic', 'StockPositions')
+                ->where('Description', config('daalder-exact.app_name'))
+                ->first();
 
-            if(is_null($stockSubscription)) {
+            if(is_null($stockSubscription) || $stockSubscription->CallbackURL !== config('app.url').'/exact/webhook/stockposition') {
                 $webhook = new WebhookSubscription($connection);
```

Comment:
Can you split this into `if(is_null(stockSubscription)) { // make new webhook }` and `if(callbackurl != config(app.url).'etc') { // update existing webhook }` (so we only check `is_null($stockSubscription)` once)? Would improve readability, I believe.